applications/RANSModellingApplication/custom_strategies/generic_residual_based_bossak_velocity_scalar_scheme.h

```diff
@@ -0,0 +1,124 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ \.
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Jordi Cotela
+//                   Suneth Warnakulasuriya (https://github.com/sunethwarna)
+//
+
+#if !defined(KRATOS_GENERIC_RESIDUAL_BASED_BOSSAK_VELOCITY_SCALAR_SCHEME_H_INCLUDED)
+#define KRATOS_GENERIC_RESIDUAL_BASED_BOSSAK_VELOCITY_SCALAR_SCHEME_H_INCLUDED
+
+// System includes
+
+// Project includes
+#include "includes/checks.h"
+#include "includes/model_part.h"
+#include "residual_based_bossak_velocity_scheme.h"
+
+// Application includes
+#include "rans_modelling_application_variables.h"
+
+namespace Kratos
+{
+template <class TSparseSpace, class TDenseSpace>
+class GenericResidualBasedBossakVelocityScalarScheme
+    : public ResidualBasedBossakVelocityScheme<TSparseSpace, TDenseSpace>
```

Comment:
to me it seems strange that the generic derives from a specific one and not the other way round

Reply:
Now using const*.
The base class is for vectors, and I needed the scalar one, so without any duplication of code, I can generate the scalar one by derriving from the vector one.